packages/@aws-cdk/aws-appmesh/README.md

```diff
@@ -1,7 +1,194 @@
+
+We recommend that you use the service discovery name of the real service that you're targeting (such as `my-service.default.svc.cluster.local`).
+
+For the provider property:
+
+* If you want the virtual service to spread traffic across multiple virtual nodes, select Virtual router and then choose the virtual router to use from the drop-down menu.
+* If you want the virtual service to reach a virtual node directly, without a virtual router, select Virtual node and then choose the virtual node to use from the drop-down menu.
+* If you don't want the virtual service to route traffic at this time (for example, if your virtual nodes or virtual router doesn't exist yet), leave blank. You can update the provider for this virtual service later.
+
+ * Adds a virtual router as the provider
+  
+```typescript
+mesh.addVirtualService('virtual-service', {
+  virtualRouterName: router.virtualRouterName,
+  virtualServiceName: 'my-service.default.svc.cluster.local',
+});
+```
+
+* Adds a virtual node as the provider
+
+```typescript
+mesh.addVirtualService('virtual-service', {
+  virtualNodeName: node.virtualNodeName,
+  virtualServiceName: `ratings.${infra.namespace.namespaceName}`,
+});
+```
+
+**Note** that only one must of `virtualNodeName` or `virtualRouterName` must be chosen.
+
+## Adding a VirtualNode
+
+A `virtual node` acts as a logical pointer to a particular task group, such as an Amazon ECS service or a Kubernetes deployment.
+
+![vn](https://docs.aws.amazon.com/app-mesh/latest/userguide/images/virtual_node.png)
+
+When you create a `virtual node`, you must specify the DNS service discovery hostname for your task group. Any inbound traffic that your `virtual node` expects should be specified as a listener. Any outbound traffic that your `virtual node` expects to reach should be specified as a backend.
+
+The response metadata for your new `virtual node` contains the Amazon Resource Name (ARN) that is associated with the `virtual node`. Set this value (either the full ARN or the truncated resource name) as the APPMESH_VIRTUAL_NODE_NAME environment variable for your task group's Envoy proxy container in your task definition or pod spec. For example, the value could be mesh/default/virtualNode/simpleapp. This is then mapped to the node.id and node.cluster Envoy parameters.
+
+> Note
+> If you require your Envoy stats or tracing to use a different name, you can override the node.cluster value that is set by APPMESH_VIRTUAL_NODE_NAME with the APPMESH_VIRTUAL_NODE_CLUSTER environment variable.
+
+```typescript
+const virtualNode = mesh.addVirtualNode('virtual-node', {
+  hostname: 'node-a',
+  namespaceName: 'default.svc.cluster.local',
+  listeners: {
+    portMappings: [
+      {
+        port: 8081,
+        protocol: appmesh.Protocol.HTTP,
+      },
+    ],
+    healthChecks: [
+      {
+        healthyThreshold: 3,
+        interval: 5000, // min
```

Comment:
Is this 5000 minutes? That's a lot, isn't it?

Generally speaking, it'd be nice to embed the unit in the key name (e.g: `intervalSeconds` or `intervalMinutes`).

Reply:
it's supposed to be a personal comment to me that's the "minimum" but forgot to remove, I will change the property to intervalMilli or similar as it's in milliseconds.

Reply:
fixed, will show on next commit